program-runtime/src/invoke_context.rs

```diff
@@ -196,7 +196,9 @@ pub struct InvokeContext<'a> {
     /// Runtime configurations used to provision the invocation environment.
     pub environment_config: EnvironmentConfig<'a>,
     /// The compute budget for the current invocation.
-    compute_budget: ComputeBudget,
+    compute_budget: SVMTransactionExecutionBudget,
+    /// The compute cost for the current invocation.
+    compute_cost: SVMTransactionExecutionCost,
```

Comment:
we already have a lifetime on `InvokeContext`, how about we just take a reference to this instead of ownership? that way we avoid copying ~300 bytes for each invoke.

Reply:
It'll be a bigger change, since a lot of test code is creating compute_budget and this instance on stack.
See `macro_rules! with_mock_invoke_context` in this file, and a bunch of other tests.

It should not be any worse than what we currently do. If we feel strongly about it, maybe a separate PR will be better.

Reply:
Separate PR is definitely fine for this. As stated, it's not worse than we have now.